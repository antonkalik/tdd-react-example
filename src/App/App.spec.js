import { render, screen } from "@testing-library/react";
import { App } from "./index";

jest.mock("src/components/Header", () => ({
  Header: () => <div data-testid="header" />,
}));

jest.mock("src/components/Body", () => ({
  Body: ({ cards }) => <div data-testid="body">{cards.length}</div>,
}));

jest.mock("src/components/Footer", () => {
  return {
    Footer: () => <div data-testid="footer" />,
  };
});

jest.mock("src/hooks/useCards", () => {
  return {
    useCards: () => ({
      cards: [],
    }),
  };
});

describe("App", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render app", () => {
    render(<App />);
    expect(screen.getByTestId("app")).toMatchSnapshot();
  });
});
