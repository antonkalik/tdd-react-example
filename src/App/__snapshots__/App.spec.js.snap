// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render app 1`] = `
<div
  class="sc-bczRLJ emciTv"
  data-testid="app"
>
  <div
    data-testid="header"
  />
  <div
    data-testid="body"
  >
    0
  </div>
  <div
    data-testid="footer"
  />
</div>
`;
